### PR TITLE
Understanding Design Patterns in Java 

### PR DESCRIPTION
**Pull request by elkiplangat**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by elkiplangat** [comment]
This fixes #1139

File: articles/design-patterns/index.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    }
+}
+```
+
+When you are using multiple threads, this style of creating a Singleton is not the best; multiple threads might get a hold of different instances of our singleton.
+There are two ways we can avoid this:
+1.  Find a way to make the getInstance method synchronized.
+2.  Create the instance of the Singleton class as soon as the class is loaded to the JVM.
+#### Limitations of the Singleton pattern
+Singletons introduce a global state. This brings about tight coupling that makes unit testing difficult.
```
Comment (botahamec) [inline articles/design-patterns/index.md:39]: You may need to briefly explain what "coupling" is. If this is meant to be read by people who don't know much about design patterns, then they probably don't know what coupling is.

**Review by botahamec — CHANGES_REQUESTED** [review]
Good first article. I only left one comment, but I just wanted to point out a couple changes that I made.

If you're showing a class name or a variable name in the article (outside a code snippet), you can a single \` to make it look like code. For example: `PhoneFactory`.

Constant names in Java are UPPER_CAMEL_CASE, so there should be underscores separating the words. I made that change to the Abstract Factory example.

Hemingway picked up a lot of hard-to-read sentences. It's good to take Hemingway's suggestions with a grain of salt, but at least look at what it's saying. It doesn't look like you've checked Grammarly either, so please do that in the future.

File: articles/design-patterns/index.md (hunk 1 above)
Comment (elkiplangat) [reply articles/design-patterns/index.md]: @botahamec  done.

**Comment by botahamec** [comment]
@hectorkambow This first-time contributor's article is ready

**Review by botahamec — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**